🔨 Add wrangler.toml file

danyx23 opened the pull request:
🔨 Add wrangler.toml file

🔨 move more settings to wrangler.toml and upgrade wrangler

danyx23 commented:
* **#3767** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3767?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>: 2  dependent PRs ([#3743](https://github.com/owid/owid-grapher/pull/3743) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3743?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>, [#3782](https://github.com/owid/owid-grapher/pull/3782) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3782?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>)
* **#3846** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3846?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danyx23 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

danyx23 commented:
I ran deployContentPreview and checked that the thumbnail worker there still does what it should: https://enable-wrangler-toml.owid-staging.pages.dev/grapher/thumbnail/life-expectancy

I think it should be safe to merge this but when we do we should be vigilant in case there is some config weirdness in prod that isn't present in staging

marcelgerber commented on wrangler.toml:
```diff
@@ -0,0 +1,23 @@
+# Generated by Wrangler on Mon Aug 05 2024 16:52:18 GMT+0000 (GMT)
+name = "owid"
+compatibility_date = "2024-04-29"
+pages_build_output_dir = "./localBake"
```
So, on `owid-cloudflare-prod` this is `~/dist`.
I think we should be fine because we're specifically running `cd dist && npx wrangler pages deploy .` in a deploy, but could you double-check?

marcelgerber commented on functions/grapher/thumbnail/[slug].ts:
```diff
@@ -6,10 +6,27 @@ export interface Env {
+                color: env.COLOR,
+                searchParams: Array.from(searchParams.entries()),
+            }
+            return new Response(JSON.stringify(info), {
+                headers: { "Content-Type": "application/json" },
+            })
```
```suggestion
            return Response.json(info)
```

danyx23 commented:
@marcelgerber I squashed the changes now of this PR and removed the debug code again. The settings from wrangler.toml showed up in the dashboard for the preview now which is nice - only the secrets are still managed via the dashboard (which makes sense). 

On staging, some settings for dev keys were not set as secrets and with the introduction of wrangler.toml they then vanished (as they are not in wrangler.toml either). Since they should have been configured as secrets anyhow, I made sure to add the dev keys as secrets for the preview deployment in the CF dashboard. On prod, everything looks good (i.e. everything is configured as secrets except for the three vars that are set in wrangler.toml.

As we discussed, since the preview deployment to CF and the prod deploy use the same mechanism and "./dist" folder, I think this should be good to go.

I'd appreciate a quick look over it as a sanity check then we can finally merge this one :)

marcelgerber reviewed (approved):
Nice, looks good!

danyx23 commented:
### Merge activity

* **Aug 13, 12:40 PM EDT**: @danyx23 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3846).
